### PR TITLE
Add chromatic, linting, TS checks

### PR DESCRIPTION
**Pull request by tmeasday**
<details>
  <summary>📦 Published PR as canary version: <code>0.0.4-canary.8.9db4163.0</code></summary>
  <br />
  
  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @chromaui/addon-visual-tests@0.0.4-canary.8.9db4163.0
  # or 
  yarn add @chromaui/addon-visual-tests@0.0.4-canary.8.9db4163.0
  ```
</details>

### CODE REVIEW

**Comment by tmeasday** [comment]
@ghengeveld @weeksling linting is failing on some of the files, I will leave to y'all to fix or change rules, I am unopinionated, I just copied the config from the CLI.

**Comment by thafryer** [comment]
Tried to rebase this PR and use PNPM since Norbert's PR got merged.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  root: true,
+  extends: ["@storybook/eslint-config-storybook"],
+  rules: {
+    "no-use-before-define": "off",
+    "@typescript-eslint/explicit-module-boundary-types": "off",
+    "eslint-comments/disable-enable-pair": "off",
+    "import/no-extraneous-dependencies": "off",
```
Comment (thafryer) [inline .eslintrc.js:8]: I don't like turning these rules off totally.
`import/no-extraneous-dependencies`
`no-use-before-define`

For no-use-before-define, I think it makes sense to configure specifically for functions and classes.
https://eslint.org/docs/latest/rules/no-use-before-define

File: package.json [hunk 2]
```diff
@@ -55,7 +56,7 @@
   },
   "dependencies": {
     "@storybook/design-system": "^7.15.11",
-    "chromatic": "^6.19.6-canary.0",
+    "chromatic": "^6.19.8-canary.2",
```
Comment (thafryer) [inline package.json:59]:
```suggestion
    "chromatic": "^6.19.8",
```

File: .eslintrc.js (hunk 1 above)
Comment (tmeasday) [reply .eslintrc.js:8]: I have no real opinions here. If y'all want to change it please do. I just copied this from the CLI.

Comment (ghengeveld) [reply .eslintrc.js]: I dropped those overrides (so they are enabled again).

**Pull request merged**